File: ScriptGraph/PinToGraph/PropertyScript/JS/LightColorProperty.js
const Amaz = effect.Amaz;

class LightColorProperty {
  constructor() {}

  setProperty(objects, property, value) {
    if (!Array.isArray(objects) || objects.length === 0) {
      return null;
    }
    objects[0][property] = new Amaz.Vector3f(value.r, value.g, value.b);
  }

  getProperty(objects, property) {
    if (!Array.isArray(objects) || objects.length === 0) {
      return null;
    }
    const vec3 = objects[0][property];
    return new Amaz.Color(vec3.x, vec3.y, vec3.z, 1.0);
  }
}

exports.LightColorProperty = LightColorProperty;

File: ScriptGraph/PinToGraph/PropertyScript/Lua/LightColorProperty.lua
local LightColorProperty = LightColorProperty or {}
LightColorProperty.__index = LightColorProperty

function LightColorProperty.new()
  local self = setmetatable({}, LightColorProperty)
  return self
end

function LightColorProperty:setProperty(objects, property, value)
  if #objects == 0 then
    return
  end
  objects[1][property] = Amaz.Vector3f(value.r, value.g, value.b)
end

function LightColorProperty:getProperty(objects, property)
  if #objects == 0 then
    return nil
  end
  local vec3 = objects[1][property]
  return Amaz.Color(vec3.x, vec3.y, vec3.z, 1.0)
end

return LightColorProperty